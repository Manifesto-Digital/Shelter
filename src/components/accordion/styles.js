import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
