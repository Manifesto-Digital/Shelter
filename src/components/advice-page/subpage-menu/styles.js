import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
