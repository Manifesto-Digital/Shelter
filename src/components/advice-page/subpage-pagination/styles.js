import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
