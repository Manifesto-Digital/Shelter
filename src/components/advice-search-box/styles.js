import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
