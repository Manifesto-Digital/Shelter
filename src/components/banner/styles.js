import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
