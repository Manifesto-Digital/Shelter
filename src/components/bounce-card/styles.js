import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
