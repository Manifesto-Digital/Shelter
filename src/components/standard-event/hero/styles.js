import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { breakpoint } from '../../theme/breakpoint';
import CTA from '../../cta';
import Button from '../../button';

export const FlexHeroContent = styled.div`
  flex: auto;
  margin-top: ${({ theme }) => theme.spacing.standard};

  ${breakpoint.desktop`
    margin-top: 0;
    flex: .35;
    flex-direction: column;
    display: flex;
  `};
`;

export const FlexHeroImage = styled.div`
  flex: auto;

  ${breakpoint.desktop`
    flex: .60;
  `};
`;

export const UnbulletedList = styled.ul`
  margin-top: ${({ theme }) => theme.spacing.small};
  list-style: none;
  padding-left: 0;
`;

export const StyledCTA = styled(CTA)`
  margin-top: auto;
`;

export const StyledButton = styled(Button)`
  margin-top: auto;
`;

export const StyledSVG = styled(SVG)`
  width: 20px;
  height: 20px;
  margin-top: -2px;
  margin-right: 10px;
  vertical-align: top;
  fill: ${({ theme }) => theme.palette.primary};
  color: ${({ theme }) => theme.palette.primary};
  display: inline-block;
`;

export const MoneySpan = styled.span`
  font-weight: bold;
  color: ${({ theme }) => theme.palette.primary};
`;

export const MapButton = styled.a`
  margin-left: ${({ theme }) => theme.spacing.small};
  padding: 5px;
  background: ${({ theme }) => theme.palette.sanMarinoBlue};
  color: white;
  cursor: pointer;

  &:hover {
    color: white;
  }
`;
