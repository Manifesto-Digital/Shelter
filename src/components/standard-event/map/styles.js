import styled from 'styled-components';

export const MapWrapper = styled.div`
  height: 300px;
  min-width: 240px;
  width: 500px;
  max-width: 100%;
`;
