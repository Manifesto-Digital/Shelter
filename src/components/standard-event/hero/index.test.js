import React from 'react';
import { shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { snapshotComponent } from 'test-helpers';
import {
  createFactory,
  createTestRef,
  createEvent,
} from '../../../utils/test-factories';
import EventHero from '.';
import { StyledButton, MapButton } from './styles';

// Default props
export const createEventHeroProps = createFactory({
  event: createEvent(),
  displayMap: true,
  mainCtaLink: 'http://example.com',
  mainCtaMethod: 'Form',
  mainCtaText: 'cta text',
  formRef: createTestRef(),
  toggleMapModal: () => {},
});

it('renders event hero correctly', () => {
  const mockData = createEventHeroProps();
  snapshotComponent(<EventHero {...mockData} />);
});

it('renders the event name as a header 1', () => {
  const event = createEvent({
    eventName: 'Amazing event',
  });
  const mockData = createEventHeroProps({ event });
  const wrapper = shallow(<EventHero {...mockData} />);
  expect(wrapper.find('h1').text()).toEqual(mockData.event.eventName);
});

it('cta button text is passed', () => {
  const event = createEvent({
    eventStatus: 'Open',
  });
  const mockData = createEventHeroProps({ event, mainCtaText: 'Open' });
  const wrapper = shallow(<EventHero {...mockData} />);

  expect(wrapper.find(StyledButton).text()).toEqual(mockData.mainCtaText);
});

it('cta button text overrides to sold out', () => {
  const event = createEvent({
    eventStatus: 'Sold Out',
  });
  const mockData = createEventHeroProps({ event, mainCtaText: 'Open' });
  const wrapper = shallow(<EventHero {...mockData} />);

  expect(wrapper.find(StyledButton).text()).toEqual('Sold Out');
});

it('trigger the show map function on click', () => {
  const mockCallback = jest.fn();
  const mockData = createEventHeroProps({ toggleMapModal: mockCallback });
  const wrapper = shallow(<EventHero {...mockData} />);

  act(() => {
    wrapper.find(MapButton).simulate('click');
  });
  expect(mockCallback).toBeCalled();
});
