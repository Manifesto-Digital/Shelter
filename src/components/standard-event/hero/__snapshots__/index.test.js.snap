// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders event hero correctly 1`] = `
.c1 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
}

.c0 {
  padding: 30px 0;
  background-color: #fff;
}

.c9 {
  display: inline-block;
  text-align: center;
  padding: 10px 1em;
  font-weight: normal;
  background-color: #ea232d;
  border: none;
  color: #fff;
  width: 100%;
}

.c9:hover {
  color: #fff;
}

.c3 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  margin-top: 20px;
}

.c10 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c4 {
  margin-top: 10px;
  list-style: none;
  padding-left: 0;
}

.c8 {
  margin-top: auto;
}

.c5 {
  width: 20px;
  height: 20px;
  margin-top: -2px;
  margin-right: 10px;
  vertical-align: top;
  fill: #ea232d;
  color: #ea232d;
  display: inline-block;
}

.c7 {
  font-weight: bold;
  color: #ea232d;
}

.c6 {
  margin-left: 10px;
  padding: 5px;
  background: #4d63b5;
  color: white;
  cursor: pointer;
}

.c6:hover {
  color: white;
}

@media (min-width:62em) {
  .c3 {
    margin-top: 0;
    -webkit-flex: .35;
    -ms-flex: .35;
    flex: .35;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:62em) {
  .c10 {
    -webkit-flex: .60;
    -ms-flex: .60;
    flex: .60;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h1>
          Brighton Half Marathon
        </h1>
        <ul
          className="c4"
        >
          <li>
            <span
              className="isvg loading c5"
            />
            Brighton
            <a
              className="c6"
              onClick={[Function]}
            >
              Map
            </a>
          </li>
          24 February 2019
          <li>
            <span
              className="isvg loading c5"
            />
            13.1 miles
          </li>
          <li>
            Registration fee:
             
            <span
              className="c7"
            >
              £
              19
            </span>
          </li>
          <li>
            Sponsorship pledge: 
            <span
              className="c7"
            >
              £
              199
            </span>
          </li>
        </ul>
        <button
          className="c8 c9"
          onClick={[Function]}
          type="button"
        >
          Sold Out
        </button>
      </div>
      <div
        className="c10"
      >
        <img
          alt="Collection hero"
          src="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?w=1000"
        />
      </div>
    </div>
  </div>
</section>
`;
